exporters/otlp/otlpmetric/otlpmetrichttp/exporter_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otlpmetrichttp // import "go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/oconf"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/internal/otest"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+func TestExporterClientConcurrentSafe(t *testing.T) {
+	const goroutines = 5
+
+	coll, err := otest.NewHTTPCollector("", nil)
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	addr := coll.Addr().String()
+	opts := []Option{WithEndpoint(addr), WithInsecure()}
+	cfg := oconf.NewHTTPConfig(asHTTPOptions(opts)...)
+	client, err := newClient(cfg)
+	require.NoError(t, err)
+
+	exp, err := newExporter(client, oconf.Config{})
+	require.NoError(t, err)
+	rm := new(metricdata.ResourceMetrics)
+
+	done := make(chan struct{})
+	first := make(chan struct{}, goroutines)
+	var wg sync.WaitGroup
+	for i := 0; i < goroutines; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			assert.NoError(t, exp.Export(ctx, rm))
+			assert.NoError(t, exp.ForceFlush(ctx))
+			// Ensure some work is done before shutting down.
+			first <- struct{}{}
+
+			for {
+				_ = exp.Export(ctx, rm)
+				_ = exp.ForceFlush(ctx)
+
+				select {
+				case <-done:
+					return
+				default:
+				}
+			}
+		}()
+	}
+
+	for i := 0; i < goroutines; i++ {
+		<-first
+	}
+	close(first)
```

Comment:
Why we are closing the channel?

1. This is not needed
2. The producer should close the channel (not the consumer).

Reply:
Similar to my response to your other comment, I had just copied this test. Please feel free to change this (and the `otlpmetricgrpc` test) in a follow up PR.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/4399